docs/scripts/buildApi.ts

```diff
@@ -221,8 +238,39 @@ async function updateStylesDefinition(context: {
       }, {} as Record<string, string>);
     }
   } catch (err) {
-    // Do nothing for now if the file doesn't exist
-    // This is still not supported for all components
+    // If the JSON file doesn't exists try extracting the info from the TS definition
```

Comment:
Is the JSON file still necessary?

Reply:
We won't after I add the `classes` prop on the `SliderStyled` too. Wanted to do the changes step by step. That will be my next PR (with that one we can drop this section and remove the JSON files

Reply:
My bad, the prop already exists there, will do the necessary updates with this PR

Reply:
Creating follow up PR for removing the JSON files

Reply:
I've decided to include the changes as part of this PR - https://github.com/mui-org/material-ui/pull/23569/commits/b254b2cd5123feb26b0bff76dacd5a2e25f8f905 as it wasn't related to the styled component.

Reply:
Fixed whitespaces too - https://github.com/mui-org/material-ui/pull/23569/commits/ec98f8d77399dae30f3ad8d6ee43b8211c4e6bd4 Seems like this is solid and we can remove the JSON files. The only info there left required was the name which can be generated easily by the filename.